### PR TITLE
279 fix incorrect return type

### LINKED ISSUE #279: Return type of scheduler::get_scheduled_times is incorrect
https://github.com/catalyst/moodle-tool_dataflows/blob/MOODLE_35_STABLE/classes/local/scheduler.php#L32-L42

Resulting in some false positives in code that uses it:

https://github.com/catalyst/moodle-tool_dataflows/blob/MOODLE_35_STABLE/classes/local/step/trigger_cron.php#L76

image

### PR DESCRIPTION
**Pull request by keevan**
- fix: return type of scheduler::get_scheduled_times
- docs: add missing param / typehints for phpdocs

Resolves #279

### CODE REVIEW

File: classes/local/step/trigger_cron.php [hunk 1]
```diff
@@ -45,9 +48,10 @@ protected static function form_define_fields(): array {
     /**
      * Get the default data.
      *
-     * @return stdClass
+     * @param \stdClass $data from the persistent form class
+     * @return \stdClass
      */
-    public function form_get_default_data(&$data) {
+    public function form_get_default_data(\stdClass &$data): \stdClass {
```
Comment (petersistrom) [inline classes/local/step/trigger_cron.php:54]: Query: A return type is defined but the function doesn't have a return statement.

Comment (keevan) [reply classes/local/step/trigger_cron.php]: Yes good catch. This was setting the values directly because it was passed in as a reference, but I think this is hard to follow so I've updated the code to always pass it by value.
